Common/Data/Custom/PsychSignal/PsychSignalSentiment.cs

```diff
@@ -29,6 +29,26 @@ namespace QuantConnect.Data.Custom.PsychSignal
     /// </summary>
     public class PsychSignalSentiment : BaseData
     {
+        /// <summary>
+        /// The EndTime of the bar represents the time the data ended and
+        /// the time it should be emitted in backtesting/live trading
+        /// </summary>
+        public override DateTime EndTime {
+            get { return Time.Add(Period).Add(FixedLiveOffset); }
+            set { Time = value.Subtract(Period); }
+        }
+
+        /// <summary>
+        /// Time from the start of the bar until the end of the bar per each data point
+        /// </summary>
+        public TimeSpan Period { get; set; } = TimeSpan.FromMinutes(1);
+
+        /// <summary>
+        /// Time it takes for data to become available on the PsychSignal API after
+        /// the time advances to the next minute
+        /// </summary>
+        public static TimeSpan FixedLiveOffset { get; set; } = TimeSpan.FromSeconds(15);
```

Comment:
Does this need to be public? This sounds like a particular implementation detail that the rest of the system should be none-the-wiser of. In fact, we would typically handle offsets like this simply via `Reader`. That being said, I do like having it explicit and the accompanying documentation. 👍  -- I think this should _minimally_ be `readonly` -- but ideally it would:

``` csharp
private static readonly TimeSpan FixedLiveOffset = TimeSpan.FromSeconds(15);
```

Also, the comments about the synchronizer's round-down behavior is tough. The issue is that the round down interval is the fill-forward interval which is dependent on _all_ subscriptions and not just this one. An interesting hack that could be done is we could provide this type a dedicated entry in `market-hours-database.json` with a time zone offset that accounts for the 15 seconds :P -- totally untested idea -- but methinks it would work and avoid coupling w/ other subscription resolution's in the algorithm.

Reply:
I believe the fill-forward interval should be provided directly from the subscription configuration https://github.com/QuantConnect/Lean/blob/master/Engine/DataFeeds/SubscriptionCollection.cs#L153 so it shouldn't be affected by this interval being directly added to the end time